crates/turbopack-css/src/module_asset.rs

```diff
@@ -118,35 +219,156 @@ impl ValueToString for ModuleChunkItem {
     async fn to_string(&self) -> Result<StringVc> {
         Ok(StringVc::cell(format!(
             "{} (css module)",
-            self.module.await?.source.path().to_string().await?
+            self.module
+                .await?
+                .inner
+                .await?
+                .source
+                .path()
+                .to_string()
+                .await?
         )))
     }
 }
 
 #[turbo_tasks::value_impl]
 impl ChunkItem for ModuleChunkItem {
     #[turbo_tasks::function]
-    fn references(&self) -> AssetReferencesVc {
-        AssetReferencesVc::cell(vec![CssProxyToCssAssetReference {
-            module: self.module,
-            context: self.context,
+    async fn references(&self) -> Result<AssetReferencesVc> {
+        // The proxy reference must come first so it is processed before other potential
+        // references inside of the CSS, like `@import` and `composes:`.
+        // This affects the order in which the resulting CSS chunks will be loaded:
+        // later references are processed first in the post-order traversal of the
+        // reference tree, and as such they will be loaded first in the resulting HTML.
+        let mut references = vec![CssProxyToCssAssetReference {
+            module: CssProxyModuleAsset {
+                module: self.module,
+            }
+            .cell(),
```

Comment:
```suggestion
            module: self.module,
```

Within `cssProxyToCssAssetReference::resolve_reference()` you can create the `CssProxyModuleAsset`